core/chains/evm/logpoller/log_poller_internal_test.go

```diff
@@ -405,43 +409,44 @@ func TestLogPoller_Replay(t *testing.T) {
 	t.Run("shutdown during replay", func(t *testing.T) {
 		lp.backupPollerNextBlock = 0
 
-		var wg sync.WaitGroup
-		wg.Add(2)
+		safeToExit := make(chan struct{})
+		pass := make(chan struct{})
 
 		ec.On("FilterLogs", mock.Anything, mock.Anything).Once().Return([]types.Log{log1}, nil).Run(func(args mock.Arguments) {
 			go func() {
-				defer wg.Done()
 				lp.replayStart <- 4
+				close(safeToExit)
 			}()
 		})
 		ec.On("FilterLogs", mock.Anything, mock.Anything).Once().Return([]types.Log{log1}, nil).Run(func(args mock.Arguments) {
-			defer wg.Done()
 			lp.cancel()
+			close(pass)
 		})
 		ec.On("FilterLogs", mock.Anything, mock.Anything).Return([]types.Log{log1}, nil).Maybe() // in case task gets delayed by >= 100ms
 
-		timeout := time.After(2 * time.Second)
+		timeLeft := testutils.WaitTimeout(t)
+		timeout := time.After(timeLeft)
 		require.NoError(t, lp.Start(tctx))
 
 		defer func() {
 			select {
 			case <-lp.replayStart: // unblock replayStart<- goroutine if it's stuck
 			default:
 			}
-			wg.Wait()
+			<-safeToExit
 			lp.Close()
 		}()
 
 		select {
 		case <-timeout:
-			assert.Fail(t, "lp.run() failed to respond to shutdown event during replay within 2s")
-		case <-utils.WaitGroupChan(&wg):
+			assert.Fail(t, fmt.Sprintf("lp.run() failed to respond to shutdown event during replay within %s", timeLeft))
```

Comment:
nit/ `assert.Failf` is available:
```suggestion
			assert.Failf(t, "lp.run() failed to respond to shutdown event during replay within %s", timeLeft)
```